### PR TITLE
Fix full source-build product build when running R2R

### PR DESCRIPTION
**Pull request by ViktorHofer**
See https://github.com/dotnet/source-build/issues/4410 When building from source, we must set the
RuntimeIdentifier property explicitly as otherwise the SDK defaults to the portable RID.

Also make the SDK aware of the R2R build by updating the Known* items in Directory.Build.targets and
remove the AspNetCore framework references.

### CODE REVIEW

**Review by vzarytovskii — APPROVED** [review]

File: buildtools/AssemblyCheck/AssemblyCheck.fsproj
Comment (tmds) [inline buildtools/AssemblyCheck/AssemblyCheck.fsproj:12]: I think these two lines can be simplified to:

```xml
<RuntimeIdentifier>$(NETCoreSdkRuntimeIdentifier)</RuntimeIdentifier>
```

Comment (ViktorHofer) [reply buildtools/AssemblyCheck/AssemblyCheck.fsproj:12]: Is the NETCoreSdkRuntimeIdentifier correct when doing a stage1 product build? Isn't that linux-x64 then?

Comment (tmds) [reply buildtools/AssemblyCheck/AssemblyCheck.fsproj]: > Is the NETCoreSdkRuntimeIdentifier correct when doing a stage1 product build and using the MSFT SDK? Isn't that linux-x64 then?

I hadn't fully considered this.

We should use `TargetRid` when we want to use what was part of the current build and meant for the target platform, and `NETCoreSdkRuntimeIdentifier` when we want what came with the build SDK and compatible with the build platform.

**Review by KevinRansom — APPROVED** [review]

**Review by vzarytovskii — APPROVED** [review]

**Comment by vzarytovskii** [comment]
Should auto-merge once approved by someone from SB-internal team and all comments are resolved.

File: buildtools/AssemblyCheck/AssemblyCheck.fsproj
Comment (ViktorHofer) [reply buildtools/AssemblyCheck/AssemblyCheck.fsproj]: TargetRid caused a build failure before:

>   /vmr/.dotnet/sdk/9.0.100-preview.5.24258.1/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Publish.targets(400,5): error NETSDK1095: Optimizing assemblies for performance is not supported for the selected target platform or architecture. Please verify you are using a supported runtime identifier, or set the PublishReadyToRun property to false. [/vmr/src/fsharp/buildtools/fslex/fslex.fsproj] [/vmr/artifacts/source-built-sdks/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj]


That's probably because the MSFT crossgen2 is built portable for linux-x64 and doesn't understand the non-portable ubuntu22.04-x64 RID. I changed `NETCoreSdkRuntimeIdentifier` everywhere which works and is correct for the msft build and the stage 2 source-build. We can revisit this if needed.

**Comment by vzarytovskii** [comment]
Oh, for some reason it wasn't blocked by the comment. @ViktorHofer is it ok that it got merged, or does it need some more changes? We can hold on with merging to release branch and wait for fixes (if any).

**Comment by ViktorHofer** [comment]
exactly

**Pull request merged**
